hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/CompressionContext.java

```diff
@@ -115,13 +114,11 @@ public void decompress(InputStream in, int inLength, byte[] outArray, int outOff
 
       // Create the input streams here the first time around.
       if (compressedIn == null) {
-        lowerIn = new BoundedDelegatingInputStream(in, inLength);
+        lowerIn = new WALDecompressionBoundedDelegatingInputStream(in);
```

Comment:
Can we guarantee that the passed `in` will not change?

Reply:
In the current implementation, it will never change. But this is a good point, there is no restriction about changing the InputStream passed in.

Let me change the code here.

Thanks.

Reply:
Fixed.